web/models/build_status_updater.ex

```diff
@@ -1,9 +1,27 @@
 defmodule Bigben.BuildStatusUpdater do
+  alias Bigben.Branch
   alias Bigben.Build
   alias Bigben.Repo
 
   def update(build_params) do
-    changeset = Build.changeset(%Build{}, build_params)
+    branch = get_branch(build_params)
+    updated_params = build_params
+      |> Map.delete(:branch)
+      |> Map.put("branch_id", branch.id)
```

Comment:
Thoughts on adding a third `branch` parameter to `Build.changeset` that accept `branch` and uses `put_assoc/3` to add the association?